src/API/Site/Controllers/Ads/CampaignController.php

```diff
@@ -192,6 +216,26 @@ protected function edit_campaign_callback(): callable {
 
 				$campaign_id = $this->ads_campaign->edit_campaign( absint( $request['id'] ), $fields );
 
+				/**
+				 * When a campaign has been successfully edited.
+				 *
+				 * @event gla_edited_campaign
+				 * @property int    id     Campaign ID.
+				 * @property string status Campaign status, `enabled` or `paused`.
+				 * @property string name   Campaign name, generated based on date.
+				 * @property float  amount Campaign budget.
+				 */
+				do_action(
+					'woocommerce_gla_track_event',
+					'edited_campaign',
+					array_merge(
+						[
+							'id' => $campaign_id,
+						],
+						$fields,
+					)
+				);
```

Comment:
What do you think to wrap these functions? So we can avoid to repeat some code 

```
public static function track_event( $event, $props ) {
do_action(
					'woocommerce_gla_track_event',
					$event,
					$props
				);
}
```

Reply:
The reason I didn't do this was because of dependencies, as well as consistency with the other tracking classes being setup to hook into an action to trigger the event.

My thought was that we would be adding tracking to many different classes, in order to be able to use a service class it would need to become a dependency of any class that wants to record a tracking event. However tracking is not a requirement for the code to function well within each of those classes, so I didn't want to add additional dependencies to multiple classes just for tracking.

The other option was to use a static class, but since the existing tracking implementations were using actions, I decided to stick with using a similar approach. The actions are also a little easier to unit test (using some helper functions) as opposed to sticking with a static class.

As for wrapping the function, the `do_action` is just one function call, so I don't see a lot of repetition there, so if a dependency is needed for a utility class I don't see us gaining a lot of benefit from wrapping it.